Shepherd should validate issue state before starting phases

## Problem

When running shepherd on issue #1725, the orchestration proceeded through curator and approval phases before failing at the builder phase with a confusing error. The issue was already closed (without a PR) and had a stale branch, but this wasn't detected until late in the process.

## Current Behavior

```
[23:11:38] [INFO] Issue: #1725
[23:14:24] [OK] Curator phase complete
[23:14:25] [OK] issue auto-approved (force mode)
[23:16:37] [ERROR] could not find PR for issue #1725
```

The shepherd wasted ~5 minutes before failing.

## Expected Behavior

Shepherd should validate issue state at the start and fail fast with a clear message:

```
[23:11:38] [INFO] Issue: #1725
[23:11:38] [ERROR] Issue #1725 is closed - cannot orchestrate
[23:11:38] [INFO] Use --reopen to reopen and continue, or check if already completed
```

## Suggested Implementation

In `orchestrate()`, before starting phases:
1. Check if issue is closed → clear error message
2. Check if existing branch `feature/issue-N` exists → warn or clean up
3. Check if PR already exists for this issue → skip to appropriate phase

## Curation Analysis

### Acceptance Criteria 1: Closed issue detection — ALREADY IMPLEMENTED

The shepherd already validates issue state in `context.py:validate_issue()` (line 164-167):
```python
state = meta.get("state", "").upper()
if state != "OPEN":
    raise IssueClosedError(issue, state)
```

This is caught in `cli.py:213-215` and returns exit code 0 with a clear message:
```
Issue #1725 is already CLOSED - no orchestration needed
```

**If the issue was truly closed and shepherd still ran curator+approval phases, that suggests the issue was open at the time of orchestration and was closed externally during the run**, or the issue was re-closed between shepherd runs.

### Acceptance Criteria 2: Stale branch warning — NOT YET IMPLEMENTED

There is no check for whether a `feature/issue-N` branch already exists on remote before creating a worktree. The `worktree.sh` script will create from the default branch, but doesn't warn about an existing stale remote branch that might cause push conflicts.

**Implementation location**: `context.py:validate_issue()` or early in `builder.py:run()`.

**Approach**: After initial validation, check for existing remote branch:
```bash
git ls-remote --heads origin feature/issue-{N}
```
If found, log a warning. This helps catch cases where a previous attempt left behind a branch.

### Acceptance Criteria 3: Existing PR detection — ALREADY IMPLEMENTED

The builder phase already checks for existing PRs in `builder.py:should_skip()` (lines 44-48):
```python
pr = get_pr_for_issue(ctx.config.issue, repo_root=ctx.repo_root)
if pr is not None:
    ctx.pr_number = pr
    return True, f"PR #{pr} already exists"
```

When a PR is found, the builder phase is skipped entirely and orchestration proceeds to the judge/merge phase.

## Revised Acceptance Criteria

Given the current implementation state, this issue should focus on the remaining gap:

- [x] ~~Shepherd fails fast with clear message when issue is closed~~ (already implemented in `context.py:164-167`)
- [ ] Warning when stale remote branch `feature/issue-N` exists before builder phase starts
- [x] ~~Detection of existing PR and skip to judge/merge phase~~ (already implemented in `builder.py:44-48`)

## Implementation Guidance

### Stale branch detection

**Where**: Add to `context.py:validate_issue()` after the state check (line 167), or as an early check in `builder.py:run()`.

**What to check**:
```bash
git ls-remote --heads origin "feature/issue-{N}"
```

**Behavior**:
- If branch exists and issue is open: log warning — `[WARN] Stale branch feature/issue-N exists on remote. Previous attempt may have left artifacts.`
- If branch exists and issue has no PR: consider whether to reuse or recreate
- In force mode: proceed with warning
- In default mode: proceed with warning (branch may be needed)

**File to modify**: `loom-tools/src/loom_tools/shepherd/context.py` (add to `validate_issue()`) or `loom-tools/src/loom_tools/shepherd/phases/builder.py` (add to `run()` before worktree creation)

### Test plan

- [ ] Unit test: stale branch detection when remote branch exists
- [ ] Unit test: no warning when no remote branch exists
- [ ] Integration: shepherd proceeds with warning when stale branch found
- [ ] Verify existing closed-issue detection still works correctly

--- a/loom-tools/src/loom_tools/shepherd/context.py
+++ b/loom-tools/src/loom_tools/shepherd/context.py
@@ -166,6 +166,9 @@ class ShepherdContext:
         if state != "OPEN":
             raise IssueClosedError(issue, state)
 
+        # Check for stale remote branch
+        self._check_stale_branch(issue)
+
         # Pre-populate label cache
         labels = {label["name"] for label in meta.get("labels", [])}
         self.label_cache.set_issue_labels(issue, labels)
@@ -180,6 +183,33 @@ class ShepherdContext:
         self.issue_title = meta.get("title", f"Issue #{issue}")
 
         return meta
+
+    def _check_stale_branch(self, issue: int) -> None:
+        """Check for existing remote branch and log a warning if found.
+
+        A stale remote branch ``feature/issue-N`` may indicate a previous
+        attempt that left artifacts behind.  We warn but always proceed
+        so orchestration is not blocked.
+        """
+        logger = logging.getLogger(__name__)
+        branch_name = f"feature/issue-{issue}"
+        try:
+            result = subprocess.run(
+                ["git", "ls-remote", "--heads", "origin", branch_name],
+                cwd=self.repo_root,
+                text=True,
+                capture_output=True,
+                check=False,
+            )
+            if result.returncode == 0 and result.stdout.strip():
+                logger.warning(
+                    "Stale branch %s exists on remote. "
+                    "Previous attempt may have left artifacts.",
+                    branch_name,
+                )
+        except OSError:
+            # git not available or other OS error — skip the check
+            pass
 
     def has_issue_label(self, label: str) -> bool:
         """Check if the issue has a specific label."""

--- a/loom-tools/tests/shepherd/test_phases.py
+++ b/loom-tools/tests/shepherd/test_phases.py
@@ -2,6 +2,8 @@
 
 from __future__ import annotations
 
+import logging
+import subprocess
 from pathlib import Path
 from unittest.mock import MagicMock, patch
 
@@ -234,3 +236,60 @@ class TestPhaseStatus:
         result = PhaseResult(status=PhaseStatus.SHUTDOWN)
         assert result.is_shutdown is True
         assert result.is_success is False
+
+
+class TestStaleBranchDetection:
+    """Test stale remote branch detection in ShepherdContext."""
+
+    def test_warns_when_stale_branch_exists(
+        self, mock_context: MagicMock, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        """Should log warning when remote branch feature/issue-N exists."""
+        ls_remote_output = "abc123\trefs/heads/feature/issue-42\n"
+        completed = subprocess.CompletedProcess(
+            args=[], returncode=0, stdout=ls_remote_output, stderr=""
+        )
+        with patch("loom_tools.shepherd.context.subprocess.run", return_value=completed):
+            with caplog.at_level(logging.WARNING, logger="loom_tools.shepherd.context"):
+                ShepherdContext._check_stale_branch(mock_context, 42)
+
+        assert any("Stale branch feature/issue-42" in r.message for r in caplog.records)
+
+    def test_no_warning_when_no_branch(
+        self, mock_context: MagicMock, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        """Should not warn when no remote branch exists."""
+        completed = subprocess.CompletedProcess(
+            args=[], returncode=0, stdout="", stderr=""
+        )
+        with patch("loom_tools.shepherd.context.subprocess.run", return_value=completed):
+            with caplog.at_level(logging.WARNING, logger="loom_tools.shepherd.context"):
+                ShepherdContext._check_stale_branch(mock_context, 42)
+
+        assert not any("Stale branch" in r.message for r in caplog.records)
+
+    def test_no_warning_on_git_failure(
+        self, mock_context: MagicMock, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        """Should silently continue when git ls-remote fails."""
+        completed = subprocess.CompletedProcess(
+            args=[], returncode=128, stdout="", stderr="fatal: error"
+        )
+        with patch("loom_tools.shepherd.context.subprocess.run", return_value=completed):
+            with caplog.at_level(logging.WARNING, logger="loom_tools.shepherd.context"):
+                ShepherdContext._check_stale_branch(mock_context, 42)
+
+        assert not any("Stale branch" in r.message for r in caplog.records)
+
+    def test_no_warning_on_os_error(
+        self, mock_context: MagicMock, caplog: pytest.LogCaptureFixture
+    ) -> None:
+        """Should silently continue when git is not available."""
+        with patch(
+            "loom_tools.shepherd.context.subprocess.run",
+            side_effect=OSError("No such file"),
+        ):
+            with caplog.at_level(logging.WARNING, logger="loom_tools.shepherd.context"):
+                ShepherdContext._check_stale_branch(mock_context, 42)
+
+        assert not any("Stale branch" in r.message for r in caplog.records)

